elytron/src/main/java/org/wildfly/extension/elytron/AuditResourceDefinitions.java

```diff
@@ -87,6 +87,11 @@
             .setRestartAllServices()
             .build();
 
+    static final SimpleAttributeDefinition AUTOFLUSH = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.AUTOFLUSH, ModelType.BOOLEAN, true)
```

Comment:
I don't think there should there be a value by default.

If `auto-flush` is not defined, it will use the value of `synchronized`(that has a default value).

if we have:

```
synchronised = false
auto-flush = <undefined>
```

the actual value of `auto-flush` will be `false` (through `synchronized`), not its own default value.

Reply:
Where are you seeing a value, this is what I see: -

    static final SimpleAttributeDefinition AUTOFLUSH = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.AUTOFLUSH, ModelType.BOOLEAN, true)
            .setAllowExpression(true)
            .setRestartAllServices()
            .build();

So it can be undefined.

Reply:
@darranl sorry for the noise,  the 3rd argument is not the default boolean value, it's for `optional`.